src/field/pokemon.ts

```diff
@@ -3581,26 +3554,18 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return false;
   }
 
-  /** @overload */
-  getTag(tagType: BattlerTagType): BattlerTag | nil;
-
-  /** @overload */
-  getTag<T extends BattlerTag>(tagType: AbstractConstructor<T>): T | nil;
-
-  getTag(tagType: BattlerTagType | AbstractConstructor<BattlerTag>): BattlerTag | nil {
+  getTag<T extends BattlerTag>(...tagTypes: BattlerTagType[]): T | nil {
     if (!this.summonData) {
       return null;
     }
-    return tagType instanceof Function
-      ? this.summonData.tags.find((t) => t instanceof tagType)
-      : this.summonData.tags.find((t) => t.tagType === tagType);
+    return this.summonData.tags.find((t) => tagTypes.includes(t.tagType)) as T | nil;
```

Comment:
```suggestion
    return this.summonData.tags.find((t) => tagTypes.includes(t.tagType)) as T | undefined;
```
It might make sense to have the `summonData` check also return `undefined`, dunno if it's useful for `getTag()` to indicate whether `summonData` is defined or not.

Reply:
I've adopted this fromt he original code:
https://github.com/Despair-Games/poketernity/blob/67d278a1b60c8d878404c72948950559076811fe/src/field/pokemon.ts#L3589C1-L3593C74

I'd rather not chnage it in the scope fo this PR

Reply:
The original code returned `undefined` here; I was talking about potentially replacing the `null` return that's above with `undefined` instead since when would it be useful to know why `getTag` didn't return the `BattlerTag` you're checking for.

Reply:
@DayKev Not sure I follow...

Reply:
https://github.com/Despair-Games/poketernity/blob/67d278a1b60c8d878404c72948950559076811fe/src/field/pokemon.ts#L3595-L3602
Originally, `getTag()` would return `null` if `pokemon.summonData` was `undefined`. If that's not the case, then `tags.find()` would return either a matching `BattlerTag`, or `undefined` if there was none. Therefore if `getTag()` returns `null` then you know that the only reason it did so was because `pokemon.summonData` was `undefined`, and if `getTag()` returned `undefined` then you know the only reason it did so is because the `BattlerTag`(s) you searched for weren't on the pokemon.

Following on from that, there are two things:
- First, I was thinking that there is probably no reason for there to be that distinction, because when calling `getTag()` you are never going to care *why* the `BattlerTag` you searched for wasn't on the pokemon, just whether it was there or not.
- Second, you changed the return type of the `tags.find()` code path from `T | undefined` (which matches what `tags.find()` will output) to `T | nil`, and my code suggestion was to return it back to `T | undefined` to match previous behavior.